sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp

```diff
@@ -324,6 +310,19 @@ namespace Azure { namespace Storage { namespace Blobs {
     Azure::Core::Response<Models::BreakBlobContainerLeaseResult> BreakLease(
         const BreakBlobContainerLeaseOptions& options = BreakBlobContainerLeaseOptions()) const;
 
+    /**
+     * @brief Marks the specified blob or snapshot for deletion. The blob is later deleted
+     * during garbage collection. Note that in order to delete a blob, you must delete all of its
+     * snapshots. You can delete both at the same time using DeleteBlobOptions.DeleteSnapshots.
+     *
+     * @param blobName The name of the blob to delete.
+     * @param options Optional parameters to execute this function.
+     * @return Nothing.
+     */
+    Azure::Core::Response<std::nullptr_t> DeleteBlob(
```

Comment:
Why are we having an inconsistent return type here?

Reply:
explained in this issue https://github.com/Azure/azure-sdk-for-cpp/issues/1322

Reply:
I created a PR #1331 to implement `Response<void>`. If core team accepts this PR, I will change the return type here.